daemon/containerd/progress.go

```diff
@@ -158,3 +159,76 @@ func (p pullProgress) UpdateProgress(ctx context.Context, ongoing *jobs, out pro
 	}
 	return nil
 }
+
+type pushProgress struct {
+	Tracker   docker.StatusTracker
+	mountable map[digest.Digest]struct{}
+	mutex     sync.Mutex
+}
+
+func (p *pushProgress) addMountable(dgst digest.Digest) {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	if p.mountable == nil {
+		p.mountable = map[digest.Digest]struct{}{}
+	}
+	p.mountable[dgst] = struct{}{}
+}
+
+func (p *pushProgress) isMountable(dgst digest.Digest) bool {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	if p.mountable == nil {
+		return false
+	}
+	_, has := p.mountable[dgst]
+	return has
+}
+
+func (p *pushProgress) UpdateProgress(ctx context.Context, ongoing *jobs, out progress.Output, start time.Time) error {
+	for _, j := range ongoing.Jobs() {
+		key := remotes.MakeRefKey(ctx, j)
+		id := stringid.TruncateID(j.Digest.Encoded())
+
+		status, err := p.Tracker.GetStatus(key)
+		if err != nil {
+			if cerrdefs.IsNotFound(err) {
+				progress.Update(out, id, "Waiting")
+				continue
+			}
+		}
+
+		if status.Committed && status.Offset >= status.Total {
+			if p.isMountable(j.Digest) {
+				progress.Update(out, id, "Mounted")
```

Comment:
I think the current output from `docker push` on a cross-blob mount specifies _where_ it was mounted from; is it possible to surface that data somehow here as well? (is that already in here and I missed it? :see_no_evil:)

Reply:
Aha, found the ref: https://github.com/moby/moby/blob/0656059ae7ab1d0b14afd3bd5f4533f2f2083ff0/distribution/push_v2.go#L379